### DISCUSSION TITLE
The principle of tid

### DISCUSSION THREAD

**Discussion by GavinZ1314** [apache/skywalking#11326] [Q&A] [answered]
What is the principle of tid, how to modify tid generation rules, and how to modify tid during code execution

**Comment by wu-sheng** [comment, marked as answer]
Trace ID implementation is determined by the agent implementation.
There is no hard requirement.
Meanwhile, most agents wouldn't provide ways to change. The source codes are licensed in APLv2, which means you can change on your own.

**Reply by GavinZ1314** [reply]
If I want to modify tid, do I need to see if the implementation in the agent can be modified independently in the project? Is there an extension interface that can be directly overwritten

**Reply by wu-sheng** [reply]
No, there isn't.

**Reply by GavinZ1314** [reply]
What is the core class for setting tid in the agent

**Reply by Superskyyy** [reply]
https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ids/GlobalIdGenerator.java here.

**Reply by GavinZ1314** [reply]
Does this look like generating tids? What I want to know is how to set tid to the current thread? I am currently not connected to Skywalking in the testing environment, but I am connected to the production environment. However, the logs are a set of configurations and use tids. I would like to simulate your agent setting tids in the testing environment

**Reply by wu-sheng** [reply]
No set is allowed.

**Reply by wu-sheng** [reply]
If you don't connect with backend, the agent stops working.

**Reply by GavinZ1314** [reply]
ok, thanks
